src/components/SearchByText/TextSearchBox.js

```diff
@@ -186,4 +198,6 @@ const mapStateToProps = createSelector(
   (main, value, pageSize) => ({ main, value, pageSize }),
 );
 
-export default connect(mapStateToProps, { goToCustomLocation })(TextSearchBox);
+export default connect(mapStateToProps, { goToCustomLocation }, null, {
+  forwardRef: true,
```

Comment:
ForwardRef here refers the passed Ref to this component right. The same thing but more in a react way?

Reply:
Sort of... not necessarily more react, but sort of force to do it that way because of the connect.
The thing is that I need the reference of the [react component](https://github.com/ProteinsWebTeam/interpro7-client/blob/45966b1f68789b3e0c1b370a867f1d951987fa9d/src/components/SearchByText/index.js#L78) to be able to call a method from it (`focus` in this case). But if I  do it the normal way with  normal `ref`, React complains that Functional components are not allowed to have references, the funny thing is that `TextSearchBox` is not funcional, it is defined using the `class` syntax. but because it is wrapped with `connect` and `connect` is now implemented using hooks, means that the wrapper is a functional component, hence the error.
So the people from `react-redux` added the ForwardRef option so they add a ForwardRef in the wrapper and then you can put  a `ref` in the component.